.github/scripts/ci_tests.sh

```diff
@@ -0,0 +1,46 @@
+#! /bin/bash
+
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+set -o errexit # Exit if command failed.
+set -o nounset # Exit if variable not set.
+set -o pipefail # Exit if pipe failed.
+
+echo -e "::group::FreeRTOS Source"
+
+if [ ! -d "libs/FreeRTOS" ]; then
+    git clone https://github.com/FreeRTOS/FreeRTOS.git libs/FreeRTOS
```

Comment:
If / when something goes wrong on a gate, how are we going to debug?  If every line - specifically like `git clone blah` - is automatically echoed then perfect before its executed, that's the best.  Otherwise we should consider adding a few `echo beginning git clone` | `starting cmake` on longer running commands that may hang or fail without flushing stdout/stderr.

Reply:
git clone does the echo for you:

https://github.com/Azure-Samples/iot-middleware-freertos-samples/runs/2760640942?check_suite_focus=true#step:5:9